### ISSUE TITLE
ACNA-697 - campaignStandardCoreAPI.getMetadataForResource does not fetch custom/extended attributes of resource

### ISSUE THREAD

**Issue by DiptimanBadajena** [adobe/aio-lib-campaign-standard#58]
While using the Campaign Standard libraries, using `getMetadataForResource(resource)` does not fetch the custom/extended attributes related to resource.

### Expected Behaviour
`getMetadataForResource(resource)` library method should pull all the attributes defined for a resource in Adobe Campaign Standard.

### Actual Behaviour
`getMetadataForResource(resource)` library method fetches the default attributes related to resource.

### Reproduce Scenario (including but not limited to)
(1) In Adobe Campaign Standard, navigate to `Custom resources` [Administration > Development > Custom resources]
(2) Click "CREATE" button and select "Extend and existing resource" (basically developers add business use case related custom/extended attributes to existing resource)
(3) Select "Profile" resource
(4) Add some simple attributes of type text / date / integer etc.
(5) Define new links to other attributes to extended resource
(6) Publish resource and the Adobe Campaign Standard forms allows users to populate the newly added attributes.

#### Steps to Reproduce
Nodesjs:
```
const acs = require('@adobe/aio-lib-campaign-standard');
const TENANT = "...";
const CLIENT_ID = "...";
const TOKEN = "...";
let get_metadata = async function (tenant, client_id, token, resource) {
    let acs_instance = await acs.init(tenant, client_id, token);
    return await acs_instance.getMetadataForResource(resource);
};
```
Invoke:
```
get_metadata(TENANT, CLIENT_ID, TOKEN, "profile").then(function (response) { console.log("success"); console.log(JSON.stringify(response))}).catch(function(error){console.log("error"); console.log(error)});
```

Response is success; however, does not show the newly added attributes in the response.
But if you refer `response.body.href` and use AJAX to retrieve the metadata; you can find all custom/extended attributes.

#### Platform and Version
```
aio -v
@adobe/aio-cli/2.3.0 darwin-x64 node-v10.16.0
```

**Comment by meryllblanchet** [comment]
Hi @DiptimanBadajena ,

Thanks for raising the issue, we'll have a look at it.

**Comment by shazron** [comment]
Investigated this. 

The url endpoint we use to get metadata only gets the standard attributes, and is defined here: https://github.com/adobe/aio-lib-campaign-standard/blob/003f4974ec8dbddb7f2738faece2d2b47c639ac1/spec/api.json#L394

If `profileAndServices` is changed to `profileAndServicesExt`, we will get the extended attributes. 

@meryllblanchet to do this change, we need to consider the impact of changing the behaviour of the API call since it will include extended attributes now that users might not expect. Is this fix considered a bug fix or a new feature? 

If it's a new feature, we could add a new parameter to the function `getMetadataForResource`, if the user wants extended attributes they would set a boolean to true, so there is no breakage -- if not, I can just patch the swagger file.

**Comment by meryllblanchet** [comment]
@shazron I like the idea of providing a new parameter to the function.

Are there any other endpoints / SDK functions that could benefit from the same strategy?

**Comment by meryllblanchet** [comment]
Fixed with #78
